Solutions/DNS Domain Solution for Microsoft Sentinel/Analytic Rules/MultipleErrorsReportedForSameDNSQueryAnomalyBased.yaml

```diff
@@ -0,0 +1,162 @@
+                    | project v = int(2)
+                ),
+                (
+                    print int(1) 
+                    | project v = print_0
+                )
+                | summarize maxv = max(v)
+                | extend sumexist = (maxv > 1)
+              );
+  let allData = union isfuzzy=true
+  (
+    (datatable(exists:int, sumexist:bool)[1,false] | where toscalar(eps) > 1000 | join (summarizationexist) on sumexist) | join (
+    _Im_Dns(starttime=todatetime(ago(2d)), endtime=now())
+    | where TimeGenerated > maxSummarizedTime and EventResultDetails in (Errors)
+    | summarize Count=count() by SrcIpAddr, DnsQuery, bin(TimeGenerated,1h)
+    | extend EventTime = TimeGenerated, Count = toint(Count), exists=int(1)
+    ) on exists
+    | project-away exists, maxv, sum*
+  ),
+  (
+    (datatable(exists:int, sumexist:bool)[1,false] | where toscalar(eps) between (501 .. 1000) | join (summarizationexist) on sumexist) | join (
+    _Im_Dns(starttime=todatetime(ago(3d)), endtime=now())
+    | where TimeGenerated > maxSummarizedTime and EventResultDetails in (Errors)
+    | summarize Count=count() by SrcIpAddr, DnsQuery, bin(TimeGenerated,1h)
+    | extend EventTime = TimeGenerated, Count = toint(Count), exists=int(1)
+    ) on exists
+    | project-away exists, maxv, sum*
+  ),
+  (
+    (datatable(exists:int, sumexist:bool)[1,false] | where toscalar(eps) <= 500 | join (summarizationexist) on sumexist) | join (
+    _Im_Dns(starttime=todatetime(ago(4d)), endtime=now())
+    | where TimeGenerated > maxSummarizedTime and EventResultDetails in (Errors)
+    | summarize Count=count() by SrcIpAddr, DnsQuery, bin(TimeGenerated,1h)
+    | extend EventTime = TimeGenerated, Count = toint(Count), exists=int(1)
+    ) on exists
+    | project-away exists, maxv, sum*
+  ),
+  (
+        DNS_Summarized_Logs_ip_CL
+        | where EventTime_t > min_t and EventResultDetails_s in (Errors)
+        | project-rename SrcIpAddr=SrcIpAddr_s, DnsQuery=DnsQuery_s, Count=count__d, EventTime=EventTime_t
+        | extend Count = toint(Count) 
+  );
+  allData
+  | make-series TotalIPCount=dcount(SrcIpAddr) on EventTime from min_t to max_t step dt by DnsQuery
+  | extend (anomalies, score, baseline) = series_decompose_anomalies(TotalIPCount, toscalar(threshold), -1, 'linefit')
+  | mv-expand anomalies, score, baseline, EventTime, TotalIPCount
+  | extend
+    anomalies = toint(anomalies),
+    score = toint(score),
+    baseline = toint(baseline),
+    EventTime = todatetime(EventTime),
+    TotalIPs = tolong(TotalIPCount)
+  | where EventTime >= ago(dt)
+  | where score >= toscalar(threshold)
+  | join kind=inner(allData | where TimeGenerated>ago(dt) | summarize SrcIps = make_set(SrcIpAddr,1000) by DnsQuery) on DnsQuery
+
+entityMappings:
+  - entityType: DNS
+    fieldMappings:
+      - identifier: DomainName
+        columnName: DnsQuery
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: SrcIps
```

Comment:
Same concern here as well. SrcIps is an array, string is expected.

Reply:
updated as previous